### PR TITLE
Add fallback constraints

### PR DESCRIPTION
**Pull request by elhesuu**
### Problem
Certain devices, mostly Android, only accept set resolutions that match exactly the supported resolution. This causes problems because we cannot find out easily what are the supported resolutions. Additionally, some devices will also fail if `width` and `height` constraints are not `Number`s.

### Solution
The proposed solution is to try fallback constraints to see if we can get the camera stream. `optional` constraints do seem to work on some devices, but not in many others. The most promising constraints seems to be omitting `width` and `height`.

### Testing results (using BrowserStack)

Devices on which this behaviour was reproduced, and can recover from the error by omitting `width` and `height`:

```
- (Android 5) Moto G 2nd Gen Chrome // optional DOES NOT work
- (Android 5) Moto X 2nd Gen Chrome // optional DOES NOT work
- (Android 6) Moto X 2nd Gen Chrome // optional DOES NOT work
- (Android 7) Galaxy S8+ Chrome // needs Number type
- (Android 7) Nexus 6p Chrome // optional DOES work
- (Android 7.1) Galaxy Note 8 Chrome // needs Number type
- (Android 7.1) Pixel Chrome // optional DOES work 
```

Devices that don't recover, even using looser constraints:
```
- (Android 4.4) Galaxy S6 Chrome
- (Android 4.4) Galaxy S6 Firefox
- (Android 5) Nexus 6 Chrome
```

Devices on which the issue was not reproduced:

```
- (Android 4.3) Galaxy Note 3 Chrome
- (Android 4.3) Galaxy Note 3 Firefox
- (Android 4.4) Galaxy Note 4 Chrome
- (Android 4.4) Galaxy Note 4 Firefox
- (Android 4.4) Galaxy S4 Chrome
- (Android 4.4) Galaxy S4 Firefox
- (Android 4.4) Galaxy S5 Chrome
- (Android 4.4) Galaxy S5 Firefox
- (Android 4.4) Galaxy Tab 4 10.1 Chrome
- (Android 4.4) Galaxy Tab 4 10.1 Firefox
- (Android 4.4) Nexus 5 Chrome
- (Android 4.4) Nexus 5 Firefox
- (Android 4.4) Nexus 7 Chrome
- (Android 5) Moto X 2nd Gen Firefox
- (Android 5) Nexus 6 Firefox
- (Android 5.1) Nexus 9 Chrome
- (Android 5.1) Nexus 9 Firefox
- (Android 5.1) Xperia Z5
- (Android 6) Galaxy S7 Chrome
- (Android 6) Galaxy S7 Firefox
- (Android 6) LG G5 Chrome
- (Android 6) LG G5 Firefox
- (Android 6) Nexus 6 Chrome
- (Android 6) Nexus 6 Firefox
- (Android 6) Nexus 7 Chrome
- (Android 6) Nexus 7 Firefox
- (Android 7) Galaxy S8 Chrome
- (Android 7) Galaxy S8 Firefox
- (Android 7) Galaxy S8+ Firefox
- (Android 7) Nexus 6p Firefox
- (Android 7) Nexus 5x (video height is 1 px)
- (Android 7.1) Galaxy Note 8 Firefox
- (Android 7.1) Pixel Firefox
- (Android 7.1) Pixel XL Chrome
- (Android 7.1) Pixel XL Firefox
- (Android 8) Galaxy S9 Chrome
- (Android 8) Galaxy S9 Firefox
- (Android 8) Galaxy S9+ Chrome
- (Android 8) Galaxy S9+ Firefox
- (Android 8) Pixel 2 Chrome
- (Android 8) Pixel 2 Firefox
- (Android 8) Pixel Chrome
- (Android 8) Pixel Firefox
```

### CODE REVIEW

File: src/react-webcam.js [hunk 1]
```diff
@@ -89,17 +91,30 @@ export default class Webcam extends Component<CameraType, State> {
...
+      facingMode,
+    }, audio};
 
     const logError = e => console.log('error', e, typeof e);
 
     const onSuccess = stream => {
       Webcam.mountedInstances.forEach((instance) => instance.handleUserMedia(stream));
     };
 
+    const fallbackConstraints = {video: true, audio };
```
Comment (rfreitas) [inline src/react-webcam.js]: hmm, not sure if I would remove `facingMode`

Comment (elhesuu) [reply src/react-webcam.js]: Absolutely @rfreitas . It shouldn't have been removed.

I added it, and tested again on affected browsers and still works well.

File: src/react-webcam.js [hunk 2]
```diff
@@ -89,17 +91,30 @@ export default class Webcam extends Component<CameraType, State> {
...
+      facingMode,
+    }, audio};
 
     const logError = e => console.log('error', e, typeof e);
 
     const onSuccess = stream => {
       Webcam.mountedInstances.forEach((instance) => instance.handleUserMedia(stream));
     };
 
+    const fallbackConstraints = { video: true, audio, facingMode };
```
Comment (rfreitas) [inline src/react-webcam.js]: isn't the facingMode inside of video? Also if we want to encode the idea of having the same constraints just without resolution constraints, we might want to do something of the sort:
```
const fallbackConstraints = { ...constraints, video: {...constraints.video, width: undefined, height: undefined} }
``` 
thoughts?

**Comment by elhesuu** [comment]
Yes, on browsers that don't remember permissions by default, they will be asked twice. 

But looking on the bright side, Firefox does not seem to fail because of constraints on any Android version/device.

**Pull request merged**